testing/python/metafunc.py

```diff
@@ -1339,6 +1346,25 @@ def test_it2(foo):
         reprec = pytester.inline_run("--collect-only")
         assert not reprec.getcalls("pytest_internalerror")
 
+    def test_parametrize_with_pathlib_object(self, pytester: Pytester) -> None:
+        """#8118"""
+        pytester.makepyfile(
+            """
+            import pathlib
+            import pytest
+
+            @pytest.mark.parametrize('file', [pathlib.Path("test1.txt")])
+            def test_runscript(file):
+                assert True
+        """
+        )
+        result = pytester.runpytest("-v", "-s")
+        result.stdout.fnmatch_lines(
+            [
+                "test_parametrize_with_pathlib_object.py::test_runscript[test1.txt] PASSED",
+            ]
+        )
+
```

Comment:
No strictly necessary as the added behaviour is covered by the previous test, but I just wanted to add a test to explicit verify the desired behaviour from the issue.